Shared/Models/Message Model/Messageable.swift

```diff
@@ -56,7 +56,7 @@ func ==(lhs: Messageable, rhs: Messageable) -> Bool {
 extension Messageable {
 
     var canBeConsumed: Bool {
-        return !self.isConsumedByMe && self.isFromCurrentUser
+        return !self.isConsumedByMe && !self.isFromCurrentUser
```

Comment:
We only want to consume messages if they're not already consumed, and they're not from the current user.